test/unit/server/middleware.server.test.ts

```diff
@@ -116,448 +113,4 @@ describe('Express Middleware', function () {
-            client.configure({
-                apiKey: 'testing',
-            })
-
-            nock.cleanAll()
-
-            const api = nock("https://api.honeybadger.io")
-                .post("/v1/notices/js")
-                .reply(201, '{"id":"1a327bf6-e17a-40c1-ad79-404ea1489c7a"}')
-
-            return new Promise<void>(resolve => {
-
-                client.beforeNotify(function (notice: Notice) {
-                    notice.context = Object.assign(notice.context, { foo: 'bar' })
-                })
-
-                client.afterNotify(function (err: Error | undefined, notice: Notice) {
-                    expect(notice.context).toEqual({ foo: 'bar' })
-                    resolve()
-                })
-
-                const callback = function(_err) {
-                    api.done()
-                }
-
-                const handler = client.lambdaHandler(function(_event, _context, _callback) {
-                    setTimeout(function() {
-                        throw new Error("Badgers!")
-                    }, 0)
-                })
-
-                handler({}, {}, callback)
-            })
-        })
-
-        it('calls callback only once if notice is sent', function () {
-            client.configure({
-                apiKey: 'testing',
-            })
-
-            nock.cleanAll()
-
-            const api = nock("https://api.honeybadger.io")
-                .post("/v1/notices/js")
-                .reply(201, '{"id":"1a327bf6-e17a-40c1-ad79-404ea1489c7a"}')
-
-            return new Promise<void>((resolve) => {
-                let callCount = 0;
-                const callback = function(_err) {
-                    callCount++
-                    setTimeout(() => {
-                        expect(callCount).toEqual(1)
-                        api.done()
-                        resolve()
-                    }, 500)
-                }
-
-                const handler = client.lambdaHandler(function(_event, _context, _callback) {
-                    setTimeout(function() {
-                        throw new Error("Badgers!")
-                    }, 0)
-                })
-
-                handler({}, {}, callback)
-            })
-        })
-
-        it('calls callback only once if preconditions fail', function () {
-            client.configure({
-                apiKey: undefined,
-            })
-
-            return new Promise<void>((resolve) => {
-                let callCount = 0;
-                const callback = function(_err) {
-                    callCount++
-                    setTimeout(() => {
-                        expect(callCount).toEqual(1)
-                        resolve()
-                    }, 500)
-                }
-
-                const handler = client.lambdaHandler(function(_event, _context, _callback) {
-                    setTimeout(function() {
-                        throw new Error("Badgers!")
-                    }, 0)
```

Comment:
The new _lambdaHandler_ implementation does not cover this case:
- An unhandled error is thrown in `setTimeout` and not passed on to the `callback` function.

However, it should be caught by the `unhandledException` plugin and report to Honeybadger as expected.

Reply:
Tested with the latest changes and it the setTimeoutError was reported as expected.